homeassistant/components/iskra/config_flow.py

```diff
@@ -0,0 +1,256 @@
+            except NotAuthorised:
+                errors["base"] = "invalid_auth"
+            except UnknownException:
+                errors["base"] = "unknown"
+
+            # if the connection was successful, create the device.
+            if not errors:
+                return await self._create_entry(
+                    self.host,
+                    self.protocol,
+                    device_info=device_info,
+                    user_input=user_input,
+                )
+
+        # If there's no user_input or there was an error, show the authentication form again.
+        return self.async_show_form(
+            step_id="authentication",
+            data_schema=STEP_AUTHENTICATION_DATA_SCHEMA,
+            errors=errors,
+        )
+
+    async def async_step_modbus_tcp(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the Modbus TCP step."""
+        errors: dict[str, str] = {}
+
+        # If there's user_input, check the connection.
+        if user_input is not None:
+            if user_input[CONF_ADDRESS] not in range(1, 256):
+                errors["base"] = "address_range"
+            else:
+                try:
+                    device_info = await test_modbus_connection(self.host, user_input)
+
+                # If the connection failed, show an error.
+                except CannotConnect:
+                    errors["base"] = "cannot_connect"
+                except UnknownException:
+                    errors["base"] = "unknown"
+
+            # If the connection was successful, create the device.
+            if not errors:
+                return await self._create_entry(
+                    host=self.host,
+                    protocol=self.protocol,
+                    device_info=device_info,
+                    user_input=user_input,
+                )
+
+        # If there's no user_input or there was an error, show the modbus form again.
+        return self.async_show_form(
+            step_id="modbus_tcp",
+            data_schema=STEP_MODBUS_TCP_DATA_SCHEMA,
+            errors=errors,
+        )
+
+    async def _create_entry(
+        self,
+        host: str,
+        protocol: str,
+        device_info: BasicInfo,
+        user_input: dict[str, Any],
+    ) -> ConfigFlowResult:
+        """Create the config entry."""
+
+        serial = device_info.serial
+        model = device_info.model
+
+        if not self.unique_id:
+            await self.async_set_unique_id(serial)
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=f"{model} {serial}",
+            data={
+                CONF_HOST: host,
+                CONF_PROTOCOL: protocol,
+                CONF_PORT: user_input.get(CONF_PORT),
+                CONF_ADDRESS: user_input.get(CONF_ADDRESS),
+                CONF_USERNAME: user_input.get(CONF_USERNAME),
+                CONF_PASSWORD: user_input.get(CONF_PASSWORD),
```

Comment:
```suggestion
                **user_input
```
I think user_input just leaves us with the rest, so we can do this. The difference in what I am suggesting is that it will omit the values that are empty. In the current code it would set the empty values to `None`

Reply:
commited suggestion, makes sense